x/umee/types/keys.go

```diff
@@ -14,9 +18,58 @@ const (
 	QuerierRoute = ModuleName
 
 	// MemStoreKey defines the in-memory store key
-	MemStoreKey = "mem_capability"
+	MemStoreKey = "mem_capability" // Question: Should this be "mem-leverage" instead?
+
+	// Prefixes to be used when storing info about individual asset or uToken types
+	// Note: The full list will likely be in ADR 003
+	AssetAssociatedUtokenPrefix = 0x01 // Stores each asset denom's accepted uToken
+	UtokenAssociatedAssetPrefix = 0x02 // Stores each uToken denom's accepted asset
+	// TODO: Add more prefixes, like AssetAssociatedOvercollatRequirement
 )
 
 func KeyPrefix(p string) []byte {
-	return []byte(p)
+	return []byte(p) // Question: Is this currently used, and should we forceASCII?
+}
+
+// returns store key used to store a specific asset coin's associated utoken denom
+func assetAssociatedUtokenKey(coin sdk.Coin) []byte {
+	return prefixDenomStoreKey(AssetAssociatedUtokenPrefix, coin)
+	// intent: store["leverage_asset_utoken_uatom"] = "u/uatom"
+}
+
+// returns store key used to store a specific utoken coin's associated asset denom
+func utokenAssociatedAssetKey(coin sdk.Coin) []byte {
+	return prefixDenomStoreKey(UtokenAssociatedAssetPrefix, coin)
+	// intent: store["leverage_utoken_asset_u/uatom"] = "uatom"
+}
+
+// prefixDenomStoreKey turns a coin to the key used to store specific info by appending a prefix to the denom.
+// Returns empty bytes on invalid denom. Modeled after x/auth/types.AccountStoreKey
+func prefixDenomStoreKey(prefix byte, coin sdk.Coin) []byte {
+	if sdk.ValidateDenom(coin.Denom) != nil {
+		// Denom did not match ^[a-z][a-z0-9/]{2,63}$
+		return []byte{}
+	}
+	// example: byte(0x01) + []byte("uatom")
+	key := []byte{prefix}
+	key = append(key, forceASCII(coin.Denom)...)
```

Comment:
```suggestion
	key = append(key, []byte(coin.Denom)...)
```
Remove forceASCII since coin denom already matches regex